Demos/DynamicLoader/Loader/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+set(SOURCES
+    main.cpp
+)
+
+#set(CMAKE_CXX_FLAGS "-fPIE ${CMAKE_CXX_FLAGS}")
+#set_target_properties(Loader PROPERTIES OUTPUT_NAME Loader)
+
+#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -nostdlib -fpic -pie")
+#set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+#set(LINKER_FLAGS "${LINKER_FLAGS} -")
+
+
+add_executable(libc.so.1 ${SOURCES})
```

Comment:
Sergey had suggested "/bin/loader" as the name, but i think it should still go in /lib. "/lib/loader" ?

Reply:
yep, /lib/loader would by my pick as well. I mean, we wouldn't want shell autocomplete to suggest running `loader` right? :laughing:

Reply:
> Sergey had suggested "/bin/loader" as the name, but i think it should still go in /lib.

But why? It's a binary isn't it?

Reply:
It's not a standard program you'll run to do some work, it's the system program interpreter. /lib or /usr/lib is also the standard location for a program loader for every SYSV following OS. The spec actually says that if you want to conform to the i386 Processor supplement, it *has* to be named /usr/lib/libc.so.1. Which, that's silly.

Reply:
Ah, that makes sense. But I don't think I agree. You're thinking of how in (at least) modern GNU/Linux, there's a distinction between how `/bin` is for tools you'd want to have in your `PATH`, and `/usr/lib` is for various system craft, including for binaries that you wouldn't run directly, but something else may need.

But we haven't been doing that in Serenity. Is, say, AudioServer a standard program you run to do some work? No, but we still put it in `/bin`.

Reply:
systemd isn't something i'd want to run manually to do some work, but it still lives in /bin on my ubuntu VM. System Software (e.g. FooServer, System Monitor, etc) for sure go in bin. Seems to me the dynamic loader is a special piece of system software that isn't a normally executed binary.

Reply:
> but it still lives in /bin on my ubuntu VM

Ubuntu must be patching it then; because [its proper location](https://github.com/systemd/systemd/blob/6bce17455e880404984761bfe064ed8ee1dce303/meson.build#L1726) is [`/usr/lib/systemd/`](https://github.com/systemd/systemd/blob/6bce17455e880404984761bfe064ed8ee1dce303/meson.build#L117)...

Reply:
> But we haven't been doing that in Serenity. Is, say, AudioServer a standard program you run to do some work? No, but we still put it in /bin.

Shouldn't it be moved, then ? It _does_ clutter the list of useful programs when autocompleting, even if it's not actually an issue yet.
Maybe there should be another folder for "service" applications which shouldn't appear in $PATH (either `/usr/lib` or a serenity-specific folder).

Reply:
I would say either we do that (create a separate location for "service" binaries) — then it would make sense to also put the dynamic loader there.

Or, and I think I prefer that, keep putting all binaries in `/bin`, libraries in `/usr/lib`, and so on.

Speaking of autocompletion, it so happens that all of our non-command-line programs start with a capital letter (examples: AudioServer, Clipboard), so they shouldn't actually clutter your autocomplete unless you intentionally type a capital letter. So how about we name the loader `/bin/Loader`?